src/tokenizer.rs

```diff
@@ -800,7 +800,9 @@ impl<'a> Tokenizer<'a> {
 
                     // mysql dialect supports identifiers that start with a numeric prefix,
                     // as long as they aren't an exponent number.
-                    if dialect_of!(self is MySqlDialect | HiveDialect) && exponent_part.is_empty() {
+                    if (dialect_of!(self is MySqlDialect | HiveDialect) && exponent_part.is_empty())
+                        || self.dialect.supports_numeric_prefix()
```

Comment:
I wonder why the check for `self.dialect.supports_numeric_prefix()` doesn't also check `exponent_part.is_empty`? 

That would mean that tokens like `1e4` are going to be parsed as identifiers

What if you wrote this check like

```rust
  if self.dialect.supports_numeric_prefix() && exponent_part.is_empty()) {
```

And then changed `MySqlDialect` and `HiveDialect` to return true for `supports_numeric_prefix`?

Reply:
Great suggestion! I overlooked this scenario, and I'll fix this logic issue